addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -304,6 +314,41 @@ def test_systray_get_activities(self):
             }
         )
 
+    def test_multi_company_notification(self):
+        # Disable the multi-company rule to collaborate between companies
+        self.env.ref('test_mail.mail_test_ticket_mc_rule').action_archive()
+        # User A in main company has a ticket and then assign it to user B in another company
+        ticket = self.env['mail.test.ticket.mc'].create({
+            'name': 'Collaborate ticket',
+            'user_id': self.user_employee.id,
+            'company_id': self.user_employee.company_id.id,
+        })
+        self.env['res.company'].invalidate_model()
```

Comment:
Did you add this because the test didn't pass without invalidation? I'm curious as to which cached field was rendered incorrect, not very important, but it's always a bit obscure to me why and when we should do this :)
I think if we can pinpoint the specific field it'd be best to explicitly pass it as a function argument.

Reply:
Yes, the field is `alias_domain_id`, inside this part:
https://github.com/odoo/odoo/blob/143cbeeba759ca3c430618c8c788c9dc6db80963/addons/mail/models/models.py#L45-L53)
In the test case, the companies has been cached from the beginning and the error will not happened. So I have to invalidate the cache.